### DISCUSSION TITLE
How do I setup pre-releases with `changeset/actions`?

### DISCUSSION THREAD

**Discussion by vicary** [changesets/changesets#1156] [Q&A]
Hello everyone, I am the maintainer of [`gqty`](https://github.com/gqty-dev/gqty). We have been using changeset for production releases for a few years.

We have been relying on `kamilkisiela/release-canary` for canary releases, but we want to replace it with the official repo `changeset/actions`.

Specifically we'd like the action to do the following:

1. We would enter prerelease mode via `changeset pre enter <tag>` if a PR calls for it. We want `changeset/actions` to publish a canary version in this format `${pre_tag}-${commit_hash}`, e.g. `gqty@3.0.0-alpha-abcdef` with the tag `alpha`, if a prerelease mode exists in the `master` branch.
2. When we merge the *Version Packages* auto-PR, automatically exit the current prerelease mode, and then publish a production version like usual.

How do I setup `changeset/actions` to do these? And how do I test my action configs without actually publishing to the current repo?

**Comment by ztamas83** [comment]
hi @vicary , did you figure this out?
